digdag-standards/src/main/java/io/digdag/standards/operator/td/TdWaitTableOperatorFactory.java

```diff
@@ -0,0 +1,184 @@
+        {
+            super(workspacePath, request);
+        }
+
+        @Override
+        public TaskResult runTask()
+        {
+            Config params = request.getConfig().mergeDefault(
+                    request.getConfig().getNestedOrGetEmpty("td"));
+
+            int pollInterval = getPollInterval(params);
+            int tableExistencePollInterval = Integer.min(pollInterval, TABLE_EXISTENCE_API_POLL_INTERVAL);
+            TableParam table = params.get("_command", TableParam.class);
+            int rows = params.get("rows", int.class, 0);
+            String engine = params.get("engine", String.class, "presto");
+
+            Config state = request.getLastStateParams().deepCopy();
+
+            try (TDOperator op = TDOperator.fromConfig(params)) {
+
+                Optional<String> existingJobId = state.getOptional(JOB_ID, String.class);
+
+                // Check if table exists using rest api (if the query job has not yet been started)
+                if (!existingJobId.isPresent()) {
+                    if (!op.tableExists(table.getTable())) {
+                        throw TaskExecutionException.ofNextPolling(tableExistencePollInterval, ConfigElement.copyOf(state));
+                    }
+
+                    // If the table exists and there's no requirement on the number of rows we're done here.
+                    if (rows <= 0) {
+                        return TaskResult.empty(request);
+                    }
+                }
+
+                // Start the query job
+                if (!existingJobId.isPresent()) {
+                    String jobId = startJob(params, table, rows, engine, op);
+                    state.set(JOB_ID, jobId);
+                    throw TaskExecutionException.ofNextPolling(JOB_STATUS_API_POLL_INTERVAL, ConfigElement.copyOf(state));
+                }
+
+                // Poll for query job status
+                assert existingJobId.isPresent();
+                TDJobOperator job = op.newJobOperator(existingJobId.get());
+                TDJob jobInfo = job.getJobInfo();
+                TDJob.Status status = jobInfo.getStatus();
+                logger.debug("poll job status: {}: {}", existingJobId.get(), jobInfo);
+
+                if (!status.isFinished()) {
+                    throw TaskExecutionException.ofNextPolling(JOB_STATUS_API_POLL_INTERVAL, ConfigElement.copyOf(state));
+                }
+
+                logger.debug("fetching poll job result: {}", existingJobId.get());
+                boolean done = fetchJobResult(rows, job);
+
+                if (!done) {
+                    state.remove(JOB_ID);
+                    throw TaskExecutionException.ofNextPolling(pollInterval, ConfigElement.copyOf(state));
+                }
+
+                return TaskResult.empty(request);
+            }
+        }
+
+        private boolean fetchJobResult(int rows, TDJobOperator job)
+        {
```

Comment:
I think here should check `status != TDJob.Status.SUCCESS` (as like TdWaitOperatorFactory.fetchJobResult does). This code throws `Got unexpected empty result for count job` which is unexpected. Here can throw TaskExecutionException with execution log of the job when the job failed.